keras/optimizers.py

```diff
@@ -69,6 +70,9 @@ def get_updates(self, params, constraints, loss):
 
     def get_gradients(self, loss, params):
         grads = K.gradients(loss, params)
+        if hasattr(self, 'normalize_gradients') and self.normalize_gradients:
+            norm = K.sqrt(sum([K.sum(K.square(g)) for g in grads])) + K.epsilon()
```

Comment:
Let's do `gradient_modifier` then (it's `modifier` so no need for `_fn`, the same way `activation` or `normalizer` and so on don't have the `_fn` suffix). Please add a test case for the use of `lambda x: l2_normalize(x)`. 

Also, feel free to add a basic MLP example script that compares no normalization, L2 normalization, and possibly a few other transformations? Entirely up to you.